crates/core/component/dex/src/component/dex.rs

```diff
@@ -83,12 +83,22 @@ impl Component for Dex {
         // For arbitrage, we extend the path search by 2 hops to allow a path out of the
         // staking token and back.
 
-        // TODO: Build an extended candidate set with:
-        // - both ends of all trading pairs for which there were swaps in the block
-        // - both ends of all trading pairs for which positions were opened
+        // Extend the fixed candidate set to include recently accessed assets, to have
+        // more arbitrage execution against newly opened positions.
+        let fixed_candidates = Arc::new(
+            routing_params
+                .fixed_candidates
+                .iter()
+                .cloned()
+                // Limit the inclusion of recently accessed assets to 10 to avoid
+                // potentially blowing up routing time.
+                .chain(state.recently_accessed_assets().iter().take(10).cloned())
```

Comment:
This could have duplicates right? should we accumulate into a BTreeSet instead until its size increases by the max limit?

Reply:
The `recently_accessed_assets` set is also incredibly likely to feature assets that are _already_ in the fixed list, perhaps they should be filtered out on insertion as well.